lib/rules/multiline-ternary.js

```diff
@@ -70,23 +55,111 @@ module.exports = {
+                                        ],
+                                        " ? "
+                                    )
+                                );
+                            }
+                        });
                     }
 
                     if (!areConsequentAndAlternateOnSameLine) {
-                        reportError(node.consequent, node, false);
+                        context.report({
+                            node: node.consequent,
+                            messageId: "unexpectedConsAlt",
+                            fix: fixer => {
+                                const consequentToken = astUtils.isParenthesised(sourceCode, node.consequent)
+                                    ? sourceCode.getTokenAfter(node.consequent)
+                                    : node.consequent;
+
+                                const alternateToken = astUtils.isParenthesised(sourceCode, node.alternate)
+                                    ? sourceCode.getTokenBefore(node.alternate)
+                                    : node.alternate;
+
+                                return (
+                                    fixer.replaceTextRange(
+                                        [
+                                            consequentToken.range[1],
+                                            alternateToken.range[0]
+                                        ],
+                                        " : "
+                                    )
+                                );
+                            }
+                        });
                     }
                 } else {
                     if (allowSingleLine && node.loc.start.line === node.loc.end.line) {
                         return;
                     }
 
                     if (areTestAndConsequentOnSameLine) {
-                        reportError(node.test, node, true);
+                        context.report({
+                            node: node.test,
+                            messageId: "expectedTestCons",
+                            fix: fixer => {
+                                const testToken = astUtils.isParenthesised(sourceCode, node.test)
+                                    ? sourceCode.getTokenAfter(node.test)
+                                    : node.test;
+
+                                const consequentToken = astUtils.isParenthesised(sourceCode, node.consequent)
+                                    ? sourceCode.getTokenBefore(node.consequent)
+                                    : node.consequent;
+
+                                return (
+                                    fixer.replaceTextRange(
+                                        [
+                                            testToken.range[1],
+                                            consequentToken.range[0]
+                                        ],
+                                        "\n? "
+                                    )
```

Comment:
Maybe we could just insert `\n` before `?` instead of replacing the range? I believe all other newline rules work that way.

(the same applies to `if (areConsequentAndAlternateOnSameLine)`)